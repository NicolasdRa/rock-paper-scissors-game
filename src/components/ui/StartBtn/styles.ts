import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    startBtn: {
      // position: "absolute",
      // bottom: "2rem",
      // right: "10rem",
      padding: ".3rem 2rem",
      color: theme.palette.common.score,
      borderColor: theme.palette.common.score,
      borderRadius: ".5rem",
      "&:hover": {
        color: theme.palette.common.white,
        borderColor: theme.palette.common.white,
      },
    },
  })
);
