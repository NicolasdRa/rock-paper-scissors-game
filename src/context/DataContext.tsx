import React, { createContext, useState } from 'react';

export type GlobalContext = {
  score: number;
  setScore: (score: number) => void;
  pick: string;
  setPick: (pick: string) => void;
};

export const DataContext = createContext<GlobalContext>({
  score: 0,
  setScore: () => {},
  pick: '',
  setPick: () => {},
});

interface Props {
  children?: React.ReactNode;
}

export const DataProvider: React.FC<Props> = ({ children }) => {
  const [score, setScore] = useState(12);
  const [pick, setPick] = useState('');

  return (
    <DataContext.Provider value={{ score, setScore, pick, setPick }}>
      {children}
    </DataContext.Provider>
  );
};
