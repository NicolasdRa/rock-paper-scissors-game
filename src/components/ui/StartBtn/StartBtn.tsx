import { useContext } from "react";
import { useHistory, useLocation } from "react-router-dom";

import { Button } from "@material-ui/core";

import { DataContext } from "../../../context/DataContext";
import { useStyles } from "./styles";

export const StartBtn = () => {
  const classes = useStyles();
  const history = useHistory();
  const location = useLocation();

  const { setScore } = useContext(DataContext);

  const startGame = () => {
    setScore(0);
    location.pathname === "/game" && history.push("/");
  };

  return (
    <Button
      onClick={() => {
        startGame();
      }}
      className={classes.startBtn}
    >
      Sart New Match
    </Button>
  );
};
