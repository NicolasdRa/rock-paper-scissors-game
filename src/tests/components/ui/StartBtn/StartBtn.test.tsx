import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { StartBtn } from "../../../../components/ui/StartBtn/StartBtn";
import { DataContext } from "../../../../context/DataContext";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const contextValue = {
  setScore: jest.fn(),
  score: 0,
  setPick: jest.fn(),
  pick: "",
};

describe("tests on <StartBtn/>", () => {
  const wrapper = mount(
    <DataContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={["/game"]}>
        <StartBtn />
      </MemoryRouter>
    </DataContext.Provider>
  );

  test("should render <StartBtn/> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should redirect to <Board/> and call setScore onClick", () => {
    wrapper.find("button").prop("onClick")!({} as any);

    expect(mockHistoryPush).toHaveBeenCalledWith("/");
    expect(contextValue.setScore).toHaveBeenCalledWith(0);
  });
});
