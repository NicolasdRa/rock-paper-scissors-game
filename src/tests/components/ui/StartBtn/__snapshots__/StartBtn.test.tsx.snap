// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <StartBtn/> should render <StartBtn/> correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-startBtn-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Sart New Match
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
